namespaces/mson-namespace.md

```diff
@@ -0,0 +1,503 @@
+        "id": "my-number"
+    },
+    "content": 42
+}
+```
+
+and can be referred to as a _base element_:
+
+```json
+{
+    "element": "my-number"
+}
+```
+
+which _expands_ to:
+
+```json
+{
+    "element": "number",
+    "attributes": {
+        "ref": "my-number"
+    },
+    "content": 42
+}
+```
+
+## Select (Distinct Element)
+
+Element representing selection of options. Every item of content array represents one possible option.
+
+### Properties
+
+- `element`: select (string, required, fixed)
+- `content` (array[Option])
+
+## Option (Distinct Element)
+
+One choice in the selection.
+
+### Properties
+
+- `element`: option (string, required, fixed)
+- `content` (enum)
+    - (Distinct Element)
+    - (array[Distinct Element])
+
+#### Example
+
+```html
+<select>
+  <option>Volvo</option>
+  <option>Saab</option>
+</select>
+```
+
+```json
+{
+    "element": "select",
+    "content": [
+        {
+            "element": "option",
+            "content": [
+                {
+                    "element": "string",
+                    "content": "volvo"
+                }
+            ]
+        }
+        {
+            "element": "option",
+            "content": [
+                {
+                    "element": "string",
+                    "content": "Saab"
+                }
+            ]
+        }
+    ]
+}
+```
+
+# MSON DOM Elements Definition
+
+## MSON Element (Distinct Element)
+
+Base element for all MSON elements.
+
+The MSON Element adds attributes representing MSON _Type Definition_ and _Type Sections_. Note in MSON DOM _Nested Member Types_ _Type Section_ is the `content` of the element.
+
+### Properties
+
+- `attributes`
+    - `typeAttributes` (array) - _Type Definition_ attributes list, see _Type Attribute_  
+        - (enum[string])
+            - required
+            - optional
+            - fixed
+            - sample
+            - default
+    - `variable` (boolean) - Element content is _Variable Value_
+    - `sample` (MSON Element) - Alternative sample _Member Types_ value
+    - `default` (MSON Element) - Default value for _Member Types_
+    - `validation` - Not used, reserved for a future use
+    - `description` - Combined description of an MSON Element
+
+## Boolean Type (JSON:Boolean Type)
+
+- Include MSON Element
+
+## String Type (JSON:String Type)
+
+- Include MSON Element
+
+## Number Type (JSON:Number Type)
+
+- Include MSON Element
+
+## Array Type (JSON:Array Type)
+
+- Include MSON Element
+
+## Object Type (JSON:Object Type)
+
+- Include MSON Element
+
+## Property Type (JSON:Property Type)
+
+- Include MSON Element
+
+## Enum Type (MSON Element)
+
+Enumeration type. Exclusive list of possible elements. The elements in content's array MUST be interpreted as mutually exclusive.
+
+### Properties
+
+- `element`: enum (string, required, fixed)
+- `content` (array[MSON Element])
+
+### Example
+
+#### MSON
+
+```
+- tag (enum[string])
+    - red
+    - green
+```
+
+#### MSON DOM
```

Comment:
What if we start using the term "refracted". That is, instead of MSON DOM, say something like _Refracted MSON_ or _Refracted JSON_ or _Refracted XML_. Just a crazy thought @zdne @smizell.